### ISSUE TITLE
[HOLD for payment 2024-02-01] [HOLD for payment 2024-01-25] [$500] Referral page - Animation on Referral page does not move

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#33046]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.12-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
1. Click on the FAB.
2. Click on "Request money"
3. Enter any amount.
4. Click on "Next".
5. Click on "Request money, get $250.".
6. Click "Learn more"

## Expected Result:
Animation on Referral page should move

## Actual Result:
Animation on Referral page does not move

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/29c2f12d-9375-4845-b006-cfe0aef9299f

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0192e339c759eff93d</li>
        <li>Upwork Job ID: 1735235226940305408</li>
        <li>Last Price Increase: 2023-12-14</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 28066892</li>
<li>BhuvaneshPatil | Contributor | 28066893</li>
</ul></li>
    </ul>
</details>

**Comment by BhuvaneshPatil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The hands on referral page are not animated

### What is the root cause of that problem?
We have used SVG icons instead of lottie aniomation.

### What changes do you think we should make in order to solve the problem?
### What alternative solutions did you explore? (Optional)
We shall use `IllustratedHeaderPageLayout` component with `LottieAnimations.Hands` similar to what we have used in `WorkspaceListPage`

**Comment by mallenexpensify** [comment]
@akinwale please review the above proposals

**Comment by akinwale** [comment]
This is a straightforward change and we can go with @ZhenjaHorbach's [proposal](https://github.com/Expensify/App/issues/33046#issuecomment-1855531002) here, considering the chronological order of the posted proposals.

🎀 👀 🎀 C+ reviewed.

**Comment by BhuvaneshPatil** [comment]
Hello @akinwale 
Can you please check the proposal that was posted early, because as. far as I remember when I posted proposal selected proposal didn't mention about `IllustratedHeaderPageLayout`. They simply mentioned about changing value of `headerContent` prop. And I agree the fix is simple so there should be emphasis on implementation details. Please re-consider.

**Comment by akinwale** [comment]
@BhuvaneshPatil 

Looking at the timestamps, it looks like both proposals were being edited simultaneously, which makes it difficult to determine who arrived at the correct solution.

In the future, it's best to follow the guidelines when making updates to proposals by creating a corresponding Updated Proposal post, so that it's easier to follow.

I will leave this up to the internal engineer to decide.

**Comment by BhuvaneshPatil** [comment]
@akinwale I understand it's quite hectic to go through timestamps.
If we consider usage of `IllustratedHeaderPageLayout`. Then my proposal was the first one to propose (in the initial version, without edit).

**Comment by rlinoz** [comment]
Looking at the timestamps the first proposal to mention IllustratedHeaderPageLayout is indeed from @BhuvaneshPatil , @akinwale are you ok if we go with his proposal?

**Comment by akinwale** [comment]
> Looking at the timestamps the first proposal to mention IllustratedHeaderPageLayout is indeed from @BhuvaneshPatil , @akinwale are you ok if we go with his proposal?

@rlinoz Yes, we can move forward with @BhuvaneshPatil proposal since it was first with the solution chronologically.

**Comment by rlinoz** [comment]
Thanks, assigned @BhuvaneshPatil to continue!

**Comment by BhuvaneshPatil** [comment]
@rlinoz @mallenexpensify 
Is this size works for lottie animation?
<img width="336" alt="Screenshot 2023-12-22 at 4 04 29 PM" src="https://github.com/Expensify/App/assets/27822551/95c7264c-5abf-41d1-8e44-408804c4720f">

**Comment by rlinoz** [comment]
@BhuvaneshPatil I am not sure what you mean

**Comment by BhuvaneshPatil** [comment]
<img width="1512" alt="Screenshot 2023-12-23 at 6 40 34 PM" src="https://github.com/Expensify/App/assets/27822551/dcdef5e5-b07d-42a9-bb86-70bbe253f4af">

After changes, this is how page looks.

**Comment by mallenexpensify** [comment]
> Auto-assign attempt failed, all eligible assignees are OOO.

hahhaaha, Happy Holidays! Let's wait a week then I'll assign someone from design to review. Cool?

**Comment by rlinoz** [comment]
Waiting holidays to re-add design label.

**Comment by mallenexpensify** [comment]
@dannymcclain 👀 please on the above for the lottie animation https://github.com/Expensify/App/issues/33046#issuecomment-1867554167

**Comment by dannymcclain** [comment]
Hmm that does not look right. I think [the latest mock](https://www.figma.com/file/V213fvgo19WofEkJYQAAX6/Referral-Program?type=design&node-id=413-2423&mode=design&t=xz3REoonHtC4d6kx-11) for this page looks like this:
![CleanShot 2024-01-03 at 08 17 48@2x](https://github.com/Expensify/App/assets/7516624/00985280-7a78-42c1-b774-0b2da7767024)

But I'm not 100% sure if we're wanting to replace the illustration Jon has in Figma with the animation. But either way, the illustration area looks much too tall and awkwardly spaced.

@shawnborton @dubielzyk-expensify do either of you have any added context on this that I may have missed?

**Comment by shawnborton** [comment]
I think this is not actually a bug, as I think this was never intended to be animated. That being said, we could totally animate it. 

I think we just need to update the illustration as Jon had mocked up (what Danny is showing). 

Either way, I don't think we should open up a new bug for proposals. This should be a follow up issue for @rezkiy37 

More context in [Slack here](https://expensify.slack.com/archives/C03TME82F/p1702569311559119)

**Comment by dubielzyk-expensify** [comment]
There wasn't any intention to animate it. Yeah, we'll just need the updated illustration

**Comment by mallenexpensify** [comment]
> I don't think we should open up a new bug for proposals. This should be a follow up issue for @rezkiy37

@shawnborton , we've already hired/assigned contributors and have an accepted proposal so it might be easiest and best to have them finish this than to ask @rezkiy37 to do so.  Either way, I think we need an updated file to use.

**Comment by shawnborton** [comment]
Works for me. Perhaps @dubielzyk-expensify can prep the new illustration to be used here.

**Comment by dubielzyk-expensify** [comment]
This should work, but let me know if you need any tweaks

[ill.svg.zip](https://github.com/Expensify/App/files/13855236/ill.svg.zip)
![ill](https://github.com/Expensify/App/assets/145416827/70754097-8509-4d63-b3f6-6ce425b5977b)

**Comment by rezkiy37** [comment]
Hey, guys! I am back. @shawnborton @mallenexpensify do you still need any my help here?

**Comment by shawnborton** [comment]
Nice, while we're at it, I think we also need to fix the text size below the headline. @dubielzyk-expensify can you provide a new mockup of what we want this to look like, and then @rezkiy37 can follow up and implement the adjustments? Thanks!

**Comment by shawnborton** [comment]
Lovely, thank you!

**Comment by rezkiy37** [comment]
Well, @dubielzyk-expensify can I ask you to share a little bit more info about the text size below the headline? I believe my eyes, but numbers are more accurate 🙂 

@shawnborton, can I use this issue for this minor fix, or will we create another one?

**Comment by rezkiy37** [comment]
Hi, I’m Michael (Mykhailo) from Callstack and I would like to work on this issue.

**Comment by dubielzyk-expensify** [comment]
Below are my measurements...
![CleanShot 2024-01-08 at 20 52 08@2x](https://github.com/Expensify/App/assets/145416827/72be633b-eddc-4eb9-89c4-5e9e8393f029)


...but, I'd lean on replicating them spacing and variables that's used on our "Workspaces" empty state as that has the sizes we wanna replicate: 
![CleanShot 2024-01-08 at 20 53 24@2x](https://github.com/Expensify/App/assets/145416827/b322e615-cfa1-4843-bbb7-0fae8cddf06a)

You can get to that screen by going to Settings/Profile icon -> Workspaces (without any workspaces set up)

**Comment by shawnborton** [comment]
I think the text size should actually be 15px, just our normal text size we use throughout the app. 

Otherwise yup, the idea is to match the workspaces page as much as possible given the layouts are nearly identical.

**Comment by rezkiy37** [comment]
@shawnborton @dubielzyk-expensify I've tried to apply the same font styles and spacings. Please check it out:

<details><summary>Dark theme</summary>

<img width="1728" alt="Dark" src="https://github.com/Expensify/App/assets/57314631/04bbad72-eb1f-4e07-a4c9-373b71b63cfe">

</details> 

<details><summary>Light theme</summary>

<img width="1728" alt="Light" src="https://github.com/Expensify/App/assets/57314631/0fcc8d08-33b8-449c-805c-1138b0f02982">

</details>

**Comment by shawnborton** [comment]
Just want to confirm that there is 20px padding both horizontally and at the top of the Title + paragraph block? It looks like the padding on the left side of the text is smaller in the left screenshot. 

For the illustration, @dubielzyk-expensify mind packaging up an svg for us when you get a minute? Thanks!

**Comment by dannymcclain** [comment]
Good catch Shawn, the padding does look a teeny bit off. Also, I think Jon already posted [the SVG](https://github.com/Expensify/App/issues/33046#issuecomment-1880210846).

**Comment by shawnborton** [comment]
Ah sweet, thanks for linking that!

**Comment by rezkiy37** [comment]
By the mockup, there is no any gap between the icon and content below. However, @dubielzyk-expensify shared with us [here](https://github.com/Expensify/App/issues/33046#issuecomment-1880210846) the icon has a space at the bottom. Can I ask you please to send one more icon without such paddings?

<img width="1080" alt="Screenshot 2024-01-08 at 15 29 55" src="https://github.com/Expensify/App/assets/57314631/36d8a0e9-1577-4587-968f-e1dd92d159a0">

**Comment by dannymcclain** [comment]
@rezkiy37 Preview/Finder does not do a good job at previewing the actual bounds of an SVG. I opened the SVG in illustrator to verify the bounding box and here is what I see:
![CleanShot 2024-01-08 at 12 15 12@2x](https://github.com/Expensify/App/assets/7516624/a8cc15d2-1709-46e1-8895-9b334f09f9f3)

In contrast, here is what I see in Finder:
![CleanShot 2024-01-08 at 12 17 29@2x](https://github.com/Expensify/App/assets/7516624/0416e45d-b10f-4ecf-be31-d3c37de25471)

And what I see in Quick Look is the same as you.

So basically, don't trust what you see in Quick Look, Finder, or Preview AT ALL for SVGs 😂. The file looks like it's good to go to me.

**Comment by rezkiy37** [comment]
Thank you, you are right!

**Comment by rezkiy37** [comment]
So, I am preparing a [draft PR](https://github.com/Expensify/App/pull/34099). You already can check screenshots out.

**Comment by BhuvaneshPatil** [comment]
Hello Everyone, I will be working on this issue, right?
I can make the changes for this issue and @rezkiy37 will continue with his changes.

**Comment by shawnborton** [comment]
Curious what @mallenexpensify thinks, but I think this is actually just follow up fixes from @rezkiy37 's original implementation, so we should not be hiring or paying anything out for this.

**Comment by mallenexpensify** [comment]
@BhuvaneshPatil at this point, since most of the work is being done already by @rezkiy37 as a followup, let's hold on having you do any work and we'll discuss once it's done (unless you're specifically asked/tagged to do work. If so, please help). 

Since you were hired, you'll be compensated, we'll discuss the percentage/amount upon completion.  That work for you?

**Comment by BhuvaneshPatil** [comment]
sure @mallenexpensify . Thanks

**Comment by rezkiy37** [comment]
The PR - https://github.com/Expensify/App/pull/34099 - is ready for review 🙂

**Comment by akinwale** [comment]
> * [x]  [@akinwale] The PR that introduced the bug has been identified. Link to the PR:
> * [x]  [@akinwale] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> * [x]  [@akinwale] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression. This was a design change.

> * [x]  [@akinwale] Determine if we should create a regression test for this bug.
> * [x]  [@akinwale] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Regression Test Steps**
1. Launch the Expensify app.
2. Open the Referral page.
3. Verify that the design matches the [mockups](https://github.com/Expensify/App/issues/33046#issuecomment-1880775380).
4. On iOS Safari, verify that the status bar background colour matches.

**Comment by akinwale** [comment]
@mallenexpensify Bumping here for payment. Thanks.

cc @BhuvaneshPatil

**Comment by mallenexpensify** [comment]
Contributor: @BhuvaneshPatil paid $500 via Upwork
Contributor+: @akinwale paid $500 via Upwork

Thanks!
